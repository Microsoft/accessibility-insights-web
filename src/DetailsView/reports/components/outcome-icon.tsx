// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedSFC } from '../../../common/react/named-sfc';
import { outcomeIconMap, OutcomeType } from './outcome-type';

interface OutcomeIconProps {
    outcomeType: OutcomeType;
}

export const OutcomeIcon = NamedSFC<OutcomeIconProps>('OutcomeIcon', props => outcomeIconMap[props.outcomeType]);
