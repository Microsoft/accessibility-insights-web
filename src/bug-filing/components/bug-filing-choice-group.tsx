// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ChoiceGroup, IChoiceGroupOption } from 'office-ui-fabric-react/lib/ChoiceGroup';
import * as React from 'react';

import { UserConfigMessageCreator } from '../../common/message-creators/user-config-message-creator';
import { NamedSFC } from '../../common/react/named-sfc';
import { BugFilingService } from '../types/bug-filing-service';

export type BugFilingChoiceGroupProps = {
    deps: BugFilingChoiceGroupDeps;
    selectedBugFilingService: BugFilingService;
    bugFilingServices: BugFilingService[];
};

export type BugFilingChoiceGroupDeps = {
    userConfigMessageCreator: UserConfigMessageCreator;
};

export const BugFilingChoiceGroup = NamedSFC<BugFilingChoiceGroupProps>('BugFilingChoiceGroup', props => {
    const { userConfigMessageCreator } = props.deps;

    const getOptions: () => IChoiceGroupOption[] = () => {
        return props.bugFilingServices.map(service => {
            return {
                key: service.key,
                text: service.displayName,
            };
        });
    };

    const onChange = (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, option?: IChoiceGroupOption) => {
        userConfigMessageCreator.setBugService(option.key);
    };

    return (
        <ChoiceGroup
            className={'bug-filing-choice-group'}
            onChange={onChange}
            options={getOptions()}
            selectedKey={props.selectedBugFilingService.key}
        />
    );
});
