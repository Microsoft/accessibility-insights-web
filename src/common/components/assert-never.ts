// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

// This function helps with exhaustiveness checking in switch statements
// If used in the default case, TS will throw an error unless all cases are handled.
export const assertNever = (x: never): never => {
    throw Error(
        'This error should never be thrown when strict null checks are applied to this project',
    );
};
