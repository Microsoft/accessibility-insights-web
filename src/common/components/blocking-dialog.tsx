// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DefaultButton, PrimaryButton } from 'office-ui-fabric-react/lib/Button';
import { Dialog, DialogFooter, DialogType, IDialogProps } from 'office-ui-fabric-react/lib/Dialog';
import * as React from 'react';

import { NamedSFC } from '../../common/react/named-sfc';

export type BlockingDialogProps = IDialogProps;

export const BlockingDialog = NamedSFC<BlockingDialogProps>('BlockingDialog', props => {
    return (
        // As of writing, NVDA has an issue where it doesn't announce new elements with role=alertdialog,
        // which Office Fabric applies implicitly whenever isBlocking: true is set. As a workaround, our
        // "blocking" dialog sets isBlocking: false (which means the dialog will use role=dialog instead of
        // role=alertdialog, which NVDA announces more reliably), and emulates being "blocking" by making
        // sure we don't set any onDismiss handlers and overriding showCloseButton to false.
        <Dialog
            {...props}
            onDismiss={undefined}
            dialogContentProps={{
                ...props.dialogContentProps,
                showCloseButton: false,
            }}
            modalProps={{
                ...props.modalProps,
                // Be warned that changing this value can change which child element is focused by default
                // (we saw this in the telemetry-permission-dialog); be sure to test this behavior specifically
                // if you ever change this back to true.
                isBlocking: false,
                onDismiss: undefined,
            }}
        />
    );
});
