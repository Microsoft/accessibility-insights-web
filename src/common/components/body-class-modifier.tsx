// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DocumentManipulator } from 'common/document-manipulator';
import * as React from 'react';

export type BodyClassModifierProps = {
    classNames: string[];
    documentManipulator: DocumentManipulator;
};

export class BodyClassModifier extends React.Component<BodyClassModifierProps> {
    public render(): JSX.Element {
        return null;
    }

    private updateClassNamesFrom(prevClassNames: string[]): void {
        const { classNames, documentManipulator } = this.props;

        const addedClassNames = classNames.filter(n => !prevClassNames.includes(n));
        const removedClassNames = prevClassNames.filter(n => !classNames.includes(n));

        // Note: may include class names not managed by us; leave them as-is
        let allBodyClassNames = documentManipulator.bodyClassNames;
        allBodyClassNames = allBodyClassNames.filter(n => !removedClassNames.includes(n));
        allBodyClassNames.push(...addedClassNames);

        documentManipulator.bodyClassNames = allBodyClassNames;
    }

    public componentDidMount(): void {
        this.updateClassNamesFrom([]);
    }

    public componentDidUpdate(prevProps: BodyClassModifierProps): void {
        this.updateClassNamesFrom(prevProps.classNames);
    }
}
