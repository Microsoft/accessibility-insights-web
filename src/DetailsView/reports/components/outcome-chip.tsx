// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../../common/react/named-sfc';
import { OutcomeIcon } from './outcome-icon';
import { OutcomeType, outcomeTypeSemantics } from './outcome-type';

interface OutcomeChipProps {
    outcomeType: OutcomeType;
    count: number;
}

export const OutcomeChip = NamedSFC<OutcomeChipProps>('OutcomeChip', props => {
    const { outcomeType, count } = props;
    const { pastTense } = outcomeTypeSemantics[outcomeType];

    const text = `${count} ${pastTense}`;

    return (
        <span className={'outcome-chip outcome-chip-' + outcomeType} title={text}>
            <span className="icon">
                <OutcomeIcon outcomeType={outcomeType} />
            </span>
            <span className="count" aria-hidden="true">
                {' '}
                {count}
            </span>
            <span className="screen-reader-only">{text}</span>
        </span>
    );
});
