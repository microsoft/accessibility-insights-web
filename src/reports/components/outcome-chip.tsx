// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';

import { OutcomeIcon } from './outcome-icon';
import { OutcomeType, outcomeTypeSemantics } from './outcome-type';

export const failureCountAutomationId = 'count';

interface OutcomeChipProps {
    outcomeType: OutcomeType;
    count: number;
    labelledBy?: boolean; // When true, suppresses the internal screen reader text
}

export const OutcomeChip = NamedFC<OutcomeChipProps>('OutcomeChip', props => {
    const { outcomeType, count, labelledBy } = props;
    const { pastTense } = outcomeTypeSemantics[outcomeType];

    const text = `${count} ${pastTense}`;

    return (
        <span className={'outcome-chip outcome-chip-' + outcomeType} title={text}>
            <span className="icon">
                <OutcomeIcon outcomeType={outcomeType} />
            </span>
            <span
                data-automation-id={failureCountAutomationId}
                className="count"
                aria-hidden="true"
            >
                {' '}
                {count}
            </span>
            {!labelledBy && <span className="screen-reader-only">{text}</span>}
        </span>
    );
});
