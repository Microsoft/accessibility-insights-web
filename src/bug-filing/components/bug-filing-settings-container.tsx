// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../common/react/named-sfc';
import { BugServiceProperties } from '../../common/types/store-data/user-configuration-store';
import { SettingsDeps } from '../../DetailsView/components/settings-panel/settings/settings-props';
import { BugFilingServiceProvider } from '../bug-filing-service-provider';
import { BugFilingService } from '../types/bug-filing-service';
import { BugFilingChoiceGroup, BugFilingChoiceGroupDeps } from './bug-filing-choice-group';

export interface BugFilingSettingsContainerProps {
    deps: BugFilingSettingsContainerDeps;
    selectedBugFilingService: BugFilingService;
    selectedBugFilingServiceData: BugServiceProperties;
}

export type BugFilingSettingsContainerDeps = {
    bugServiceProvider: BugFilingServiceProvider;
} & BugFilingChoiceGroupDeps &
    SettingsDeps;

export const BugFilingSettingsContainer = NamedSFC<BugFilingSettingsContainerProps>('BugFilingSettingsContainer', props => {
    const { deps, selectedBugFilingService, selectedBugFilingServiceData } = props;
    const SettingsForm = selectedBugFilingService.renderSettingsForm;
    const bugFilingServices = deps.bugServiceProvider.allVisible();

    return (
        <>
            <BugFilingChoiceGroup deps={deps} selectedBugFilingService={selectedBugFilingService} bugFilingServices={bugFilingServices} />
            <SettingsForm deps={deps} settings={selectedBugFilingServiceData} />
        </>
    );
});
