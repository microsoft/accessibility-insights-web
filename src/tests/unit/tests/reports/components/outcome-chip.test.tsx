// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import * as React from 'react';
import { OutcomeChip } from 'reports/components/outcome-chip';
import { OutcomeIcon } from 'reports/components/outcome-icon';
import { mockReactComponents } from 'tests/unit/mock-helpers/mock-module-helpers';

jest.mock('reports/components/outcome-icon');

describe('OutcomeChip', () => {
    mockReactComponents([OutcomeIcon]);
    describe('render', () => {
        test('pass', () => {
            const renderResult = render(<OutcomeChip outcomeType="pass" count={3} />);
            expect(renderResult.asFragment()).toMatchSnapshot();
        });
        test('incomplete', () => {
            const renderResult = render(<OutcomeChip outcomeType="incomplete" count={2} />);
            expect(renderResult.asFragment()).toMatchSnapshot();
        });
        test('fail', () => {
            const renderResult = render(<OutcomeChip outcomeType="fail" count={4} />);
            expect(renderResult.asFragment()).toMatchSnapshot();
        });

        test('inapplicable', () => {
            const renderResult = render(<OutcomeChip outcomeType="inapplicable" count={4} />);
            expect(renderResult.asFragment()).toMatchSnapshot();
        });
    });
});
