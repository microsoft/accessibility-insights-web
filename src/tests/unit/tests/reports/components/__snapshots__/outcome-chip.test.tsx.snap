// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutcomeChip render fail 1`] = `
<DocumentFragment>
  <span
    class="outcome-chip outcome-chip-fail"
    title="4 Failed"
  >
    <span
      class="icon"
    >
      <mock-outcomeicon
        outcometype="fail"
      />
    </span>
    <span
      aria-hidden="true"
      class="count"
      data-automation-id="count"
    >
       4
    </span>
    <span
      class="screen-reader-only"
    >
      4 Failed
    </span>
  </span>
</DocumentFragment>
`;

exports[`OutcomeChip render inapplicable 1`] = `
<DocumentFragment>
  <span
    class="outcome-chip outcome-chip-inapplicable"
    title="4 Not applicable"
  >
    <span
      class="icon"
    >
      <mock-outcomeicon
        outcometype="inapplicable"
      />
    </span>
    <span
      aria-hidden="true"
      class="count"
      data-automation-id="count"
    >
       4
    </span>
    <span
      class="screen-reader-only"
    >
      4 Not applicable
    </span>
  </span>
</DocumentFragment>
`;

exports[`OutcomeChip render incomplete 1`] = `
<DocumentFragment>
  <span
    class="outcome-chip outcome-chip-incomplete"
    title="2 Incomplete"
  >
    <span
      class="icon"
    >
      <mock-outcomeicon
        outcometype="incomplete"
      />
    </span>
    <span
      aria-hidden="true"
      class="count"
      data-automation-id="count"
    >
       2
    </span>
    <span
      class="screen-reader-only"
    >
      2 Incomplete
    </span>
  </span>
</DocumentFragment>
`;

exports[`OutcomeChip render pass 1`] = `
<DocumentFragment>
  <span
    class="outcome-chip outcome-chip-pass"
    title="3 Passed"
  >
    <span
      class="icon"
    >
      <mock-outcomeicon
        outcometype="pass"
      />
    </span>
    <span
      aria-hidden="true"
      class="count"
      data-automation-id="count"
    >
       3
    </span>
    <span
      class="screen-reader-only"
    >
      3 Passed
    </span>
  </span>
</DocumentFragment>
`;
